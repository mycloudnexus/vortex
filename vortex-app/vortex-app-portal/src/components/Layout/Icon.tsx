export const DefaultLogo = () => (
  <svg width='36' height='36' viewBox='0 0 36 36' fill='none' xmlns='http://www.w3.org/2000/svg'>
    <g clipPath='url(#clip0_2848_34972)'>
      <path
        d='M13.1289 2.24829C16.1289 1.37329 23.4789 0.823293 28.8789 5.62329C35.6289 11.6233 34.5039 20.2484 33.7539 22.8734'
        stroke='url(#paint0_linear_2848_34972)'
        strokeWidth='2.6'
        strokeLinecap='round'
      />
      <path
        d='M10.3804 13.4768L17.9712 9.06871L25.5404 13.4394L25.5188 22.2182L17.928 26.6264L10.3588 22.2557L10.3804 13.4768Z'
        fill='#AABFFA'
      />
      <path
        d='M3.62288 8.28134L3.63171 4.69301L6.73443 2.8912L9.82162 4.67385L9.81279 8.26217L6.71006 10.064L3.62288 8.28134Z'
        stroke='#AABFFA'
        strokeWidth='1.55594'
      />
      <path
        d='M26.1386 27.7055L30.4036 25.2287L34.6566 27.6845L34.6444 32.617L30.3794 35.0938L26.1265 32.638L26.1386 27.7055Z'
        fill='#BB5EF1'
      />
      <path
        d='M22.6992 34.1189C19.6992 34.9939 12.3492 35.5439 6.94922 30.7439C0.19922 24.7439 1.32422 16.1188 2.07422 13.4938'
        stroke='url(#paint1_linear_2848_34972)'
        strokeWidth='2.6'
        strokeLinecap='round'
      />
    </g>
    <defs>
      <linearGradient
        id='paint0_linear_2848_34972'
        x1='23.7272'
        y1='1.66406'
        x2='23.7272'
        y2='22.8734'
        gradientUnits='userSpaceOnUse'
      >
        <stop stopColor='#BBCDFF' />
        <stop offset='1' stopColor='#D0AEFF' />
      </linearGradient>
      <linearGradient
        id='paint1_linear_2848_34972'
        x1='12.1009'
        y1='34.7031'
        x2='12.1009'
        y2='13.4938'
        gradientUnits='userSpaceOnUse'
      >
        <stop stopColor='#95B1FF' />
        <stop offset='1' stopColor='#C398FF' />
      </linearGradient>
      <clipPath id='clip0_2848_34972'>
        <rect width='36' height='36' fill='white' />
      </clipPath>
    </defs>
  </svg>
)
export const DefaultCompanyLogo = () => (
  <svg xmlns='http://www.w3.org/2000/svg' width='16' height='18' viewBox='0 0 16 18' fill='none'>
    <path
      d='M9.28857 2.01046V15.649H10.3071V15.629L12.6567 15.6295V7.6157L10.3071 6.79808V5.26475L13.6129 6.41427C13.7559 6.50576 13.8758 6.62911 13.9632 6.77468C14.0506 6.92024 14.1031 7.08407 14.1167 7.25332L14.12 7.33903V15.649H15.2381V17.0967H0V15.6495H1.11238V4.58284C1.11238 4.20284 1.31095 3.84951 1.63667 3.64808L7.58286 1.0757C8.32572 0.61665 9.28857 1.14427 9.28857 2.01046ZM7.82524 2.54998L2.57619 4.82094V15.6305H7.82524V9.59094L7.81572 9.59475V8.92522L7.82524 8.94808V6.16379L7.81572 6.1676V5.50094L7.82524 5.52236V2.54998ZM6.83333 8.43141V9.99665L3.56952 11.329V9.76379L6.83333 8.43141ZM6.83333 5.00903V6.58094L3.56952 7.95427V6.38236L6.83333 5.00951V5.00903Z'
      // fill='black'
      fill='currentColor'
      fillOpacity='0.45'
    />
  </svg>
)
