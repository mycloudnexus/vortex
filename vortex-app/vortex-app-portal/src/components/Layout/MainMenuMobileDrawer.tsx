import { Drawer, Menu } from 'antd'
import * as styles from './index.module.scss'
import { useAppStore } from '@/stores/app.store'
import { styled } from 'styled-components'

type Props = {
  open: boolean
  onClose: () => void
  items: any
  activeKeys: string[]
  setActiveKeys: (activeKeys: string[]) => void
}

const MainMenuMobileDrawer = ({ open, onClose, items, activeKeys, setActiveKeys }: Props) => {
  const { mainColor } = useAppStore()
  const MenuCustom = styled(Menu)`
    .ant-menu-submenu-selected {
      svg {
        path {
          stroke: ${mainColor};
        }
      }
    }
    .ant-menu-item-selected {
      svg {
        path {
          fill: ${mainColor};
          stroke: ${mainColor};
        }
        circle {
          stroke: ${mainColor};
        }
      }
    }
  `
  return (
    <Drawer
      headerStyle={{ display: 'none' }}
      rootClassName={styles.mobileDrawer}
      open={open}
      width='100vw'
      onClose={onClose}
      placement='left'
      mask={false}
    >
      <MenuCustom
        onSelect={(e) => {
          setActiveKeys(e.selectedKeys)
          onClose()
        }}
        className={styles.menu}
        selectedKeys={activeKeys}
        mode='inline'
        items={items}
      />
    </Drawer>
  )
}

export default MainMenuMobileDrawer
