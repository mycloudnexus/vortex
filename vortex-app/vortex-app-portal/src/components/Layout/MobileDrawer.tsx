import { Avatar, Col, Drawer, Flex, Row } from 'antd'
import * as styles from './index.module.scss'
import { Fragment } from 'react'
import { CloseOutlined, LogoutOutlined } from '@ant-design/icons'

type Props = {
  open: boolean
  onClose: () => void
}

const MobileDrawer = ({ open, onClose }: Props) => {
  const menus = [
    { title: 'Tim Tim', onClick: () => {}, icon: <Avatar size='small'>TP</Avatar> },
    { title: 'Dashboard', onClick: () => {} },
    { title: 'Logout', onClick: () => {}, icon: <LogoutOutlined style={{ color: '#fff', cursor: 'pointer' }} /> }
  ]
  return (
    <Drawer
      headerStyle={{ display: 'none' }}
      className={styles.drawer}
      open={open}
      width={`calc(100vw - 145px)`}
      onClose={onClose}
    >
      <Row style={{ width: '100%' }}>
        {menus.map((item, index) => (
          <Fragment key={item.title}>
            <Col span={3}>
              <div className={styles.menuIcon}>{item.icon}</div>
            </Col>
            <Col span={21}>
              <Flex align='center' justify='space-between' className={styles.menuItem}>
                <div className={index !== 0 ? styles.avatarItem : styles.userName}>{item.title}</div>
                {index === 0 && <CloseOutlined style={{ color: '#fff', cursor: 'pointer' }} onClick={onClose} />}
              </Flex>
            </Col>
          </Fragment>
        ))}
      </Row>
    </Drawer>
  )
}

export default MobileDrawer
