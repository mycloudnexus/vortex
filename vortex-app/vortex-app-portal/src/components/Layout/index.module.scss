.appWrapper {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 999;
  min-height: 100vh;

  // Tablet
  @media only screen and (min-width: 320px) {
    :global {
      .headroom {
        padding: 10px 20px;
        background: #1a263d;
      }
    }
  }
}

.container {
  height: calc(100vh - 78px);

  :global {
    .ant-layout-sider-children {
      height: calc(100% + 200px);
    }
  }
}

.pageNav {
  h3 {
    color: #fff;
  }
}

.settingDropdown {
  :global {
    .ant-dropdown-menu {
      background: #16263f;
    }
    .ant-dropdown-menu-title-content {
      color: #fff;
      &:hover {
        color: #23e0f9;
      }
    }
  }
}

.avatarMenu {
  background: #16263f;
  padding: 10px;
  width: 235px;
}

.userName {
  font-size: 14px;
  color: #fff;
}

.divider {
  height: 1px;
  width: 100%;
  background: #fff;
  opacity: 0.2;
  margin: 15px 0;
}

.avatarItem {
  width: 100%;
  color: #fff;
  cursor: pointer;
  &:hover {
    color: #23e0f9;
  }
}

.menuMobile {
  cursor: pointer;
}

.drawer {
  :global {
    .ant-drawer-body {
      padding: 10px;
    }
    .ant-drawer-body {
      background: #1a263d;
    }
  }
}

.menuItem {
  border-bottom: 1px solid hsla(0, 0%, 100%, 0.1);
  padding: 0.5rem 0;
}

.menuIcon {
  margin-top: 0.3rem;
  cursor: pointer;
}

.slider {
  :global {
    .ant-layout-sider-trigger {
      display: none;
    }
  }
}

.network {
  padding: 8px 18px 12px;
}

.collapseBtn {
  svg {
    cursor: pointer;
    color: #fff;
    font-size: 9px;
  }
}

.menu {
  padding-top: 8px;
  height: 100%;
  background: #f2f5f6;
  :global {
    .ant-menu-item {
      margin: 0;
      width: 100%;
    }
    .ant-menu-item-selected {
      background-color: #fff;
    }
    .ant-menu-submenu-title {
      padding-right: 13px;
      margin: 0 !important;
      width: 100%;
    }
  }
}

.mobileDrawer {
  :global {
    .ant-drawer-body {
      padding: 0;
    }
    .ant-drawer-content-wrapper {
      top: 127px;
    }
  }
}
