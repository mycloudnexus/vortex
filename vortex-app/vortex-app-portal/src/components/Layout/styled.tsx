import Sider from 'antd/es/layout/Sider'
import { styled } from 'styled-components'
const SliderCustom = (mainColor: string) => styled(Sider)`
  .ant-menu-submenu-selected {
    .icon-dashboard {
      path {
        fill: ${mainColor};
      }
    }
    .icon-dc {
      path {
        stroke: ${mainColor};
      }
    }
    .icon-l2 {
      path {
        fill: ${mainColor};
      }
    }
    .icon-l3 {
      path {
        stroke: ${mainColor};
      }
      circle {
        stroke: ${mainColor};
      }
    }
    .icon-cr {
      path:not(.special-path) {
        stroke: ${mainColor};
      }
      .special-path {
        fill: ${mainColor};
      }
    }
    .icon-setting {
      path {
        fill: ${mainColor};
      }
    }
  }
  .ant-menu-item-selected {
    .icon-dashboard {
      path {
        fill: ${mainColor};
      }
    }
    .icon-dc {
      path {
        stroke: ${mainColor};
      }
    }
    .icon-l2 {
      path {
        fill: ${mainColor};
      }
    }
    .icon-l3 {
      path {
        stroke: ${mainColor};
      }
      circle {
        stroke: ${mainColor};
      }
    }
    .icon-cr {
      path:not(.special-path) {
        stroke: ${mainColor};
      }
      .special-path {
        fill: ${mainColor};
      }
    }
    .icon-setting {
      path {
        fill: ${mainColor};
      }
    }
  }
`

export { SliderCustom }
