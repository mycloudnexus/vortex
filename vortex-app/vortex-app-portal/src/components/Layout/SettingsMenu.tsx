import useDeviceDetect from '@/hooks/useDeviceDetect'
import { Avatar, Dropdown, Flex } from 'antd'
import Text from '../Text'
import { useAppStore } from '@/stores/app.store'
import { DownOutlined, MenuOutlined } from '@ant-design/icons'
import * as styles from './index.module.scss'
import { useState } from 'react'
import MobileDrawer from './MobileDrawer'

const SettingsMenu = () => {
  const { isMobile } = useDeviceDetect()
  const { currentCompany, currentUser } = useAppStore()
  const [openDrawer, setOpenDrawer] = useState(false)
  const items = [
    {
      label: 'po',
      key: '0',
      onClick: () => {}
    },
    {
      label: 'ping',
      key: '1',
      onClick: () => {}
    }
  ]
  const dropdownRender = () => {
    return (
      <div className={styles.avatarMenu}>
        <Flex justify='space-between'>
          <Flex vertical gap={10}>
            <div className={styles.userName}>{currentUser?.name}</div>
            <div className={styles.avatarItem}>Personal details</div>
          </Flex>
          <Avatar size='large'>TP</Avatar>
        </Flex>
        <div className={styles.divider} />
        <div className={styles.avatarItem}>Logout</div>
      </div>
    )
  }
  return (
    <Flex align='center'>
      {openDrawer && <MobileDrawer open={openDrawer} onClose={() => setOpenDrawer(false)} />}
      {isMobile ? (
        <div className={styles.menuMobile}>
          <MenuOutlined style={{ color: '#fff' }} onClick={() => setOpenDrawer(true)} />
        </div>
      ) : (
        <Flex align='center' id='nav' gap={24} justify='flex-end'>
          <Flex gap={8}>
            <Text.LightMedium color='rgba(255, 255, 255, 0.50)'>Viewing as</Text.LightMedium>
            <Dropdown
              menu={{ items }}
              getPopupContainer={() => document.getElementById('nav') as HTMLDivElement}
              overlayClassName={styles.settingDropdown}
            >
              <Flex gap={4}>
                <Text.LightMedium color='#fff'>{currentCompany?.name}</Text.LightMedium>
                <DownOutlined style={{ color: '#fff', fontSize: 10 }} />
              </Flex>
            </Dropdown>
          </Flex>
          <Dropdown
            menu={{ items: [] }}
            getPopupContainer={() => document.getElementById('nav') as HTMLDivElement}
            dropdownRender={dropdownRender}
          >
            <Avatar size='large'>TP</Avatar>
          </Dropdown>
        </Flex>
      )}
    </Flex>
  )
}

export default SettingsMenu
