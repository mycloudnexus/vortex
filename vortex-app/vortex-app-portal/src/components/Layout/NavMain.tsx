import { Link } from 'react-router-dom'
import SettingsMenu from './SettingsMenu'
import { Flex } from 'antd'
import * as styles from './index.module.scss'

const NavMain = () => {
  const userIsLoggedIn = true
  return (
    <Flex justify='space-between' align='center'>
      <nav>
        <Flex align='center' gap={16} className={styles.pageNav}>
          <Link to={''}>
            <img alt='Vortex' src={''} />
          </Link>
          <Link to={''}>
            <h3>Vortex</h3>
          </Link>
        </Flex>
      </nav>

      <nav data-iht='user-nav' className='user-nav'>
        {userIsLoggedIn && <SettingsMenu />}
      </nav>
    </Flex>
  )
}

export default NavMain
