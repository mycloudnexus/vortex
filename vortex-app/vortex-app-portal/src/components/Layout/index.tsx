import { Suspense, useEffect, useState } from 'react'
import { Outlet, useLocation, useNavigate } from 'react-router-dom'
import Headroom from 'react-headroom'
import NavMain from './NavMain'
import * as styles from './index.module.scss'
import Sider from 'antd/es/layout/Sider'
import { Flex, Menu } from 'antd'
import NetIcon from '@/assets/icon/network.svg'
import { useAppStore } from '@/stores/app.store'
import { DoubleLeftOutlined, DoubleRightOutlined, DownOutlined, RightOutlined } from '@ant-design/icons'
import { useBoolean } from 'usehooks-ts'
import Text from '../Text'
import { ReactComponent as DashboardIcon } from '@/assets/icon/dashboard.svg'
import { ReactComponent as DCIcon } from '@/assets/icon/dcport.svg'
import { ReactComponent as CRIcon } from '@/assets/icon/cloudrouter.svg'
import { ReactComponent as L2Icon } from '@/assets/icon//l2.svg'
import { ReactComponent as L3Icon } from '@/assets/icon/l3.svg'
import { ReactComponent as SettingIcon } from '@/assets/icon/setting.svg'
import { styled } from 'styled-components'
import useDeviceDetect from '@/hooks/useDeviceDetect'
import MainMenuMobileDrawer from './MainMenuMobileDrawer'
import Authenticate from '../Access/Authenticate'

const Layout = () => {
  const { mainColor } = useAppStore()
  const SliderCustom = styled(Sider)`
    .ant-menu-submenu-selected {
      svg {
        path {
          stroke: ${mainColor};
        }
      }
    }
    .ant-menu-item-selected {
      svg {
        path {
          fill: ${mainColor};
          stroke: ${mainColor};
        }
        circle {
          stroke: ${mainColor};
        }
      }
    }
  `
  const { isMobile } = useDeviceDetect()
  const [activeKeys, setActiveKeys] = useState(['1'])
  const { value: collapsed, setValue: setCollapsed, setTrue: trueCollapse, setFalse: falseCollapse } = useBoolean(false)
  const { value: mainMobileDrawer, toggle: toggleDrawer, setFalse: falseDrawer } = useBoolean(false)
  const location = useLocation()
  const [openKeys, setOpenKeys] = useState<string[]>()
  const navigate = useNavigate()
  useEffect(() => {
    if (!isMobile) {
      falseDrawer()
    }
  }, [isMobile])

  useEffect(() => {
    if (location.pathname === '/') {
      setActiveKeys(['1'])
      return
    }
    for (const m of items) {
      if (m?.regex?.test(location.pathname)) {
        setActiveKeys([m.key])
      }
      if (m.children) {
        for (const c of m.children) {
          if ((c as any).regex?.test(location.pathname)) {
            setActiveKeys([m.key, c.key])
          }
        }
      }
    }
  }, [location])

  const items = [
    {
      key: '1',
      icon: <DashboardIcon />,
      label: 'Dashboard',
      regex: /^\/dashboard(\/.*)?$/,
      onClick: () => {
        navigate('/')
      }
    },
    {
      key: '2',
      icon: <DCIcon />,
      label: 'DC Ports',
      children: [{ key: '2-1', label: 'View all' }]
    },
    {
      key: '3',
      icon: <L2Icon />,
      label: 'L2 Connections'
    },
    {
      key: '4',
      icon: <L3Icon />,
      label: 'L3 connections'
    },
    {
      key: '5',
      icon: <CRIcon />,
      label: 'CloudRouter'
    },
    {
      key: '6',
      icon: <SettingIcon />,
      label: 'Settings'
    }
  ]

  return (
    <div className={styles.appWrapper}>
      <Authenticate>
        <MainMenuMobileDrawer
          onClose={falseDrawer}
          open={mainMobileDrawer}
          items={items}
          activeKeys={activeKeys}
          setActiveKeys={setActiveKeys}
        />
        <Headroom disableInlineStyles>
          <Suspense fallback=''>
            <NavMain />
          </Suspense>
        </Headroom>
        <Flex vertical={isMobile} className={styles.container}>
          {isMobile ? (
            <Flex
              role='none'
              style={{ background: mainColor, cursor: 'pointer' }}
              className={styles.network}
              gap={12}
              align='center'
              onClick={toggleDrawer}
            >
              <img src={NetIcon} alt='network' />
              {!collapsed && <Text.NormalLarge color='#fff'>NETWORK</Text.NormalLarge>}
            </Flex>
          ) : (
            <SliderCustom collapsible collapsed={collapsed} onCollapse={setCollapsed} className={styles.slider}>
              <Flex vertical style={{ background: mainColor }} className={styles.network}>
                <Flex justify='flex-end' className={styles.collapseBtn}>
                  {!collapsed ? (
                    <DoubleLeftOutlined onClick={trueCollapse} role='none' />
                  ) : (
                    <DoubleRightOutlined onClick={falseCollapse} role='none' />
                  )}
                </Flex>
                <Flex gap={12} align='center' style={{ marginTop: 16 }}>
                  <img src={NetIcon} alt='network' />
                  {!collapsed && <Text.NormalLarge color='#fff'>NETWORK</Text.NormalLarge>}
                </Flex>
              </Flex>
              <Menu
                openKeys={openKeys}
                onOpenChange={(k) => setOpenKeys(k)}
                onSelect={(e) => {
                  setActiveKeys(e.selectedKeys)
                }}
                className={styles.menu}
                selectedKeys={activeKeys}
                mode='inline'
                items={items}
                expandIcon={(iconInfo) => (iconInfo.isOpen ? <DownOutlined /> : <RightOutlined />)}
              />
            </SliderCustom>
          )}
          <Outlet />
        </Flex>
      </Authenticate>
    </div>
  )
}

export default Layout
