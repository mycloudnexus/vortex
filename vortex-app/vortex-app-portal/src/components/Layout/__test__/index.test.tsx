import { render, act } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { QueryClient, QueryClientProvider } from 'react-query'
import Layout from '..'
import * as Auth0 from '@auth0/auth0-react'
import * as userHooks from '@/hooks/user'

jest.mock('@/utils/helpers/request')
const mockedUsedNavigate = jest.fn()
const queryClient = new QueryClient()

jest.mock('@auth0/auth0-react')
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate
}))
jest.mock('@/stores/app.store', () => ({
  useAppStore: () => {
    return {
      userType: 'reseller',
      currentAuth0User: {},
      setRoleList: jest.fn(),
      setDownstreamUser: jest.fn(),
      setCustomerUser: jest.fn(),
      setCustomerCompanies: jest.fn()
    }
  }
}))

jest.mock('@/assets/icon/dashboard.svg', () => ({
  ReactComponent: () => <svg />
}))
jest.mock('@/assets/icon/dcport.svg', () => ({
  ReactComponent: () => <svg />
}))
jest.mock('@/assets/icon/cloudrouter.svg', () => ({
  ReactComponent: () => <svg />
}))
jest.mock('@/assets/icon//l2.svg', () => ({
  ReactComponent: () => <svg />
}))
jest.mock('@/assets/icon/l3.svg', () => ({
  ReactComponent: () => <svg />
}))
jest.mock('@/assets/icon/setting.svg', () => ({
  ReactComponent: () => <svg />
}))
jest
  .spyOn(Auth0, 'useAuth0')
  .mockReturnValue({ isAuthenticated: true, isLoading: false, user: {}, getAccessTokenSilently: jest.fn() } as any)

const BaseLayout = () => (
  <QueryClientProvider client={queryClient}>
    <BrowserRouter>
      <Layout />
    </BrowserRouter>
  </QueryClientProvider>
)

describe('Layout component with data', () => {
  const testName = 'test'
  const testEmail = 'test@test.com'
  beforeEach(() => {
    jest.spyOn(userHooks, 'useGetUserAuthDetail').mockReturnValue({
      data: {
        data: {
          name: testName,
          email: testEmail
        }
      }
    } as any)
  })
  it('save user data after get data ', async () => {
    await act(async () => render(<BaseLayout />))
    const userName = window.portalLoggedInUser.name
    const userEmail = window.portalLoggedInUser.email
    expect(userName).toEqual(testName)
    expect(userEmail).toEqual(testEmail)
  })
})
// describe('Layout component', () => {
//   beforeEach(() => {
//     jest
//       .spyOn(Auth0, 'useAuth0')
//       .mockReturnValue({ isAuthenticated: true, isLoading: false, user: {}, getAccessTokenSilently: jest.fn() } as any)
//   })
//   it('renders the layout component with authenticated ', async () => {
//     const { container } = await act(async () => render(<BaseLayout />))
//     expect(container.firstChild).toMatchSnapshot()
//   })
// })
