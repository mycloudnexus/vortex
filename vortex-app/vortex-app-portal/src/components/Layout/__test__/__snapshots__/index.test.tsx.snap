// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders the layout component with authenticated  1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-center ant-flex-justify-space-between"
      >
        <nav>
          <div
            class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-center"
            style="gap: 16px;"
          >
            <a
              href="/"
            >
              <svg
                fill="none"
                height="36"
                viewBox="0 0 36 36"
                width="36"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_2848_34972)"
                >
                  <path
                    d="M13.1289 2.24829C16.1289 1.37329 23.4789 0.823293 28.8789 5.62329C35.6289 11.6233 34.5039 20.2484 33.7539 22.8734"
                    stroke="url(#paint0_linear_2848_34972)"
                    stroke-linecap="round"
                    stroke-width="2.6"
                  />
                  <path
                    d="M10.3804 13.4768L17.9712 9.06871L25.5404 13.4394L25.5188 22.2182L17.928 26.6264L10.3588 22.2557L10.3804 13.4768Z"
                    fill="#AABFFA"
                  />
                  <path
                    d="M3.62288 8.28134L3.63171 4.69301L6.73443 2.8912L9.82162 4.67385L9.81279 8.26217L6.71006 10.064L3.62288 8.28134Z"
                    stroke="#AABFFA"
                    stroke-width="1.55594"
                  />
                  <path
                    d="M26.1386 27.7055L30.4036 25.2287L34.6566 27.6845L34.6444 32.617L30.3794 35.0938L26.1265 32.638L26.1386 27.7055Z"
                    fill="#BB5EF1"
                  />
                  <path
                    d="M22.6992 34.1189C19.6992 34.9939 12.3492 35.5439 6.94922 30.7439C0.19922 24.7439 1.32422 16.1188 2.07422 13.4938"
                    stroke="url(#paint1_linear_2848_34972)"
                    stroke-linecap="round"
                    stroke-width="2.6"
                  />
                </g>
                <defs>
                  <lineargradient
                    gradientUnits="userSpaceOnUse"
                    id="paint0_linear_2848_34972"
                    x1="23.7272"
                    x2="23.7272"
                    y1="1.66406"
                    y2="22.8734"
                  >
                    <stop
                      stop-color="#BBCDFF"
                    />
                    <stop
                      offset="1"
                      stop-color="#D0AEFF"
                    />
                  </lineargradient>
                  <lineargradient
                    gradientUnits="userSpaceOnUse"
                    id="paint1_linear_2848_34972"
                    x1="12.1009"
                    x2="12.1009"
                    y1="34.7031"
                    y2="13.4938"
                  >
                    <stop
                      stop-color="#95B1FF"
                    />
                    <stop
                      offset="1"
                      stop-color="#C398FF"
                    />
                  </lineargradient>
                  <clippath
                    id="clip0_2848_34972"
                  >
                    <rect
                      fill="white"
                      height="36"
                      width="36"
                    />
                  </clippath>
                </defs>
              </svg>
            </a>
            <a
              href="/"
            >
              <h3>
                Console Connect Private Label
              </h3>
            </a>
          </div>
        </nav>
        <nav
          class="user-nav"
          data-iht="user-nav"
        >
          <div
            class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-center"
          >
            <div
              class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-center ant-flex-justify-flex-end"
              id="nav"
              style="color: red; gap: 24px;"
            >
              <span
                class="ant-avatar ant-avatar-lg ant-avatar-circle ant-avatar-image ant-dropdown-trigger css-dev-only-do-not-override-49qm"
              >
                <img
                  alt="avatar"
                />
              </span>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </div>
  <div
    class="ant-flex css-dev-only-do-not-override-49qm"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark ant-layout-sider-has-trigger sc-dprtRQ dUEafh css-dev-only-do-not-override-49qm"
      style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-stretch ant-flex-vertical"
        >
          <div
            class="ant-flex css-dev-only-do-not-override-49qm ant-flex-justify-flex-end"
          >
            <span
              aria-label="double-left"
              class="anticon anticon-double-left"
              role="none"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                data-icon="double-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M272.9 512l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L186.8 492.3a31.99 31.99 0 000 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H532c6.7 0 10.4-7.7 6.3-12.9L272.9 512zm304 0l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L490.8 492.3a31.99 31.99 0 000 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H836c6.7 0 10.4-7.7 6.3-12.9L576.9 512z"
                />
              </svg>
            </span>
          </div>
          <div
            class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-center"
            style="margin-top: 16px; gap: 12px;"
          >
            <img
              alt="network"
              src="[object Object]"
            />
            <span
              style="font-size: 16px; font-weight: 500; color: rgb(255, 255, 255); font-style: normal;"
            >
              NETWORK
            </span>
          </div>
        </div>
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light css-dev-only-do-not-override-49qm"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-menu-item ant-menu-item-selected"
            data-menu-id="rc-menu-uuid-test-1"
            regex="/^\\/dashboard(\\/.*)?$/"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <svg />
            <span
              class="ant-menu-title-content"
            >
              Dashboard
            </span>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="none"
          >
            <div
              aria-controls="rc-menu-uuid-test-2-popup"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              data-menu-id="rc-menu-uuid-test-2"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <svg />
              <span
                class="ant-menu-title-content"
              >
                DC Ports
              </span>
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-3"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <svg />
            <span
              class="ant-menu-title-content"
            >
              L2 Connections
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-4"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <svg />
            <span
              class="ant-menu-title-content"
            >
              L3 connections
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-5"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <svg />
            <span
              class="ant-menu-title-content"
            >
              CloudRouter
            </span>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="none"
          >
            <div
              aria-controls="rc-menu-uuid-test-6-popup"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              data-menu-id="rc-menu-uuid-test-6"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <svg />
              <span
                class="ant-menu-title-content"
              >
                Settings
              </span>
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
      <div
        class="ant-layout-sider-trigger"
        style="width: 200px;"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
      </div>
    </aside>
    <div
      class="ant-flex css-dev-only-do-not-override-49qm ant-flex-align-start ant-flex-justify-start ant-flex-vertical"
      style="width: 100%;"
    >
      <nav
        class="ant-breadcrumb sc-dstKZu cCCkCF css-dev-only-do-not-override-49qm"
        style="font-size: 18px;"
      >
        <ol>
          <li>
            <a
              class="sc-khLCKb eYFRrJ"
              href="/"
            >
              Dashboard
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
</div>
`;
